CHANGES.txt

```diff
@@ -45,6 +45,9 @@ Breaking Changes
 Changes
 =======
 
+- Fixed and issue that could cause incorrect results for a ``join`` query with
```

Comment:
I thought this only affected master -> so no changes entry would be necessary as the bug isn't present in released versions?

Reply:
@mfussenegger @matriv I thought the push down optimisation was in 2.3 already no?

Reply:
@andreidan For Union yes, not for joins...

Reply:
@mfussenegger created https://github.com/crate/crate/pull/6921 to drop the changes and add some motivation in the LogicalPlanBuilder documentation